src/main/java/org/tinyradius/util/RadiusServer.java

```diff
@@ -353,15 +372,15 @@ protected void listen(DatagramSocket s) {
 				// check client
 				InetSocketAddress localAddress = (InetSocketAddress) s.getLocalSocketAddress();
 				InetSocketAddress remoteAddress = new InetSocketAddress(packetIn.getAddress(), packetIn.getPort());
-				String secret = getSharedSecret(remoteAddress);
+				String secret = getSharedSecret(remoteAddress, makeRadiusPacket(packetIn, "1234567890", RadiusPacket.RESERVED));
```

Comment:
If you follow my suggestion above, this would be:

```
RadiusPacket packet = makeRadiusPacket(packetIn, "1234567890");
packet.setPacketType(RadiusPacket.RESERVED);
String secret = getSharedSecret(remoteAddress, packet);
```

I don't understand the reason for the hardcoded value "1234..." though.  Could you explain?